lib/Dialect/ESI/runtime/cosim_dpi_server/DpiEntryPoints.cpp

```diff
@@ -17,21 +17,27 @@
 //===----------------------------------------------------------------------===//
 
 #include "dpi.h"
+#include "esi/Context.h"
 #include "esi/Ports.h"
 #include "esi/backends/RpcServer.h"
 
 #include <algorithm>
 #include <cassert>
 #include <cstdlib>
+#include <format>
 
 using namespace esi;
 using namespace esi::cosim;
 
 /// If non-null, log to this file. Protected by 'serverMutex`.
 static FILE *logFile;
+static std::unique_ptr<Context> context = nullptr;
```

Comment:
We may have a teardown memory issue here. If the context gets destroyed before server and server attempts to log a message, but things could happen. This was part of the reason it's easier for Context to own everything and not rely on C++ destruction order. I think this is the correct order so that server gets deconstructed first, but you may want to note this in the code.